### PR TITLE
EDX_BLND_CLI-203 Library Authoring - Edit Library Title

### PR DESCRIPTION
**Pull request by UvgenGen**
[EDX_BLND_CLI-203](https://youtrack.raccoongang.com/agiles/104-969/105-2602?issue=EDX_BLND_CLI-203) Library Authoring - Edit Library Title

### CODE REVIEW

File: src/library-authoring/author-library/LibraryAuthoringPage.jsx [hunk 1]
```diff
@@ -324,6 +330,75 @@ const deriveTypeOptions = (blockTypes, intl) => {
...
+    setIsActive(true);
+  };
+
+  return (
+    <>
+      <small className="card-subtitle">{intl.formatMessage(messages['library.detail.page.heading'])}</small>
+      <h1 className="page-header-title">
+        {inputIsActive
+          ? (
+            <>
```
Comment (ruzniaievdm) [inline src/library-authoring/author-library/LibraryAuthoringPage.jsx]: This brackets is redundant, remove please

File: src/library-authoring/author-library/specs/LibraryAuthoringPage.spec.jsx [hunk 2]
```diff
@@ -323,4 +326,23 @@ testSuite('<LibraryAuthoringPageContainer />', () => {
...
     );
   });
+
+  it('Rename library', async () => {
+    const library = libraryFactory();
+    const block = blockFactory(undefined, { library });
+    await render(library, genState(library, [block]));
+    
+    const editButton = screen.getByTestId('edit-name-button');
+    act(() => {
```
Comment (ruzniaievdm) [inline src/library-authoring/author-library/specs/LibraryAuthoringPage.spec.jsx]: Wrapping action in `act` unnecessarily

File: src/library-authoring/author-library/specs/LibraryAuthoringPage.spec.jsx [hunk 3]
```diff
@@ -323,4 +326,23 @@ testSuite('<LibraryAuthoringPageContainer />', () => {
...
+      editButton.click();
+    });
+    const input = await screen.getByTestId('library-name-input');
+    act(() => {
```
Comment (ruzniaievdm) [inline src/library-authoring/author-library/specs/LibraryAuthoringPage.spec.jsx]: here too

File: src/library-authoring/author-library/specs/LibraryAuthoringPage.spec.jsx [hunk 4]
```diff
@@ -323,4 +326,23 @@ testSuite('<LibraryAuthoringPageContainer />', () => {
       () => expect(deleteLibraryBlock.fn).toHaveBeenCalledWith({ blockId: block.id }),
     );
   });
+
+  it('Rename library', async () => {
+    const library = libraryFactory();
+    const block = blockFactory(undefined, { library });
+    await render(library, genState(library, [block]));
+    
+    const editButton = screen.getByTestId('edit-name-button');
```
Comment (ruzniaievdm) [inline src/library-authoring/author-library/specs/LibraryAuthoringPage.spec.jsx]: More preferable to use `getByRole` instead `getByTestId`
`const editButton = screen.getByRole('button', { name: /Edit/i });` 
just update alt attr 'Edit' to something more specific, like `edit-name-button` and remove `data-testid`

File: src/library-authoring/author-library/specs/LibraryAuthoringPage.spec.jsx [hunk 5]
```diff
@@ -323,4 +326,23 @@ testSuite('<LibraryAuthoringPageContainer />', () => {
...
+    act(() => {
+      editButton.click();
+    });
+    const input = await screen.getByTestId('library-name-input');
```
Comment (ruzniaievdm) [inline src/library-authoring/author-library/specs/LibraryAuthoringPage.spec.jsx]: here too

**Comment by ruzniaievdm** [comment]
@UvgenGen left small discussions, please take a look

File: src/library-authoring/author-library/LibraryAuthoringPage.jsx (hunk 1 above)
Comment (UvgenGen) [reply src/library-authoring/author-library/LibraryAuthoringPage.jsx]: Updated

File: src/library-authoring/author-library/specs/LibraryAuthoringPage.spec.jsx (hunk 2 above)
Comment (UvgenGen) [reply src/library-authoring/author-library/specs/LibraryAuthoringPage.spec.jsx]: Updated

File: src/library-authoring/author-library/specs/LibraryAuthoringPage.spec.jsx (hunk 3 above)
Comment (UvgenGen) [reply src/library-authoring/author-library/specs/LibraryAuthoringPage.spec.jsx]: Updated

File: src/library-authoring/author-library/specs/LibraryAuthoringPage.spec.jsx (hunk 4 above)
Comment (UvgenGen) [reply src/library-authoring/author-library/specs/LibraryAuthoringPage.spec.jsx]: Updated

File: src/library-authoring/author-library/specs/LibraryAuthoringPage.spec.jsx (hunk 5 above)
Comment (UvgenGen) [reply src/library-authoring/author-library/specs/LibraryAuthoringPage.spec.jsx]: Updated

File: src/library-authoring/author-library/specs/LibraryAuthoringPage.spec.jsx [hunk 6]
```diff
@@ -323,4 +326,19 @@ testSuite('<LibraryAuthoringPageContainer />', () => {
       () => expect(deleteLibraryBlock.fn).toHaveBeenCalledWith({ blockId: block.id }),
     );
   });
+
+  it('Rename library', async () => {
+    const library = libraryFactory();
+    const block = blockFactory(undefined, { library });
+    await render(library, genState(library, [block]));
+    
+    const editButton = screen.getByRole('button', { name: /edit name button/i })
```
Comment (ruzniaievdm) [inline src/library-authoring/author-library/specs/LibraryAuthoringPage.spec.jsx:335]: nit: its working, but why do you use lower case `/edit name button/I`? `alt="Edit name button"` is written with capitalise letter.
Its little bit confusing.

also below

**Review by ruzniaievdm — APPROVED** [review]

File: src/library-authoring/author-library/LibraryAuthoringPage.jsx [hunk 7]
```diff
@@ -324,6 +330,71 @@ const deriveTypeOptions = (blockTypes, intl) => {
...
+
+/**
+ * LibraryAuthoringPageHeaderBase
+ * Title component for the LibraryAuthoringPageBase.
+ */
+const LibraryAuthoringPageHeaderBase = ({intl, library, ...props}) => {
+  const [inputIsActive, setIsActive] = useState(false);
+  const handleOnBlur = (event) => {
+    var newTitle = event.target.value;
+    if (newTitle && newTitle != library.title){
```
Comment (monteri) [inline src/library-authoring/author-library/LibraryAuthoringPage.jsx]: `if (newTitle && newTitle != library.title) {` space after condition before curly brace

**Review by monteri — APPROVED** [review]

**Pull request merged**